pcdsdevices/delay_generator.py

```diff
@@ -0,0 +1,133 @@
+"""
+Delay generator class
+
+This module contains classes related to the SRS DG645 delay generator.
+"""
+
+from ophyd import Device
+from ophyd import Component as Cpt
+from ophyd import EpicsSignalRO
+from ophyd import EpicsSignal
+from ophyd.utils.epics_pvs import set_and_wait
+
+from .interface import BaseInterface
+
+
+CHANNELS = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'T0']
+TRIGGER_SOURCES = {
+    '0': 'Internal',
+    '1': 'Ext ^edge',
+    '2': 'Ext ~edge',
+    '3': 'SS ext ^edge',
+    '4': 'SS ext ~edge',
+    '5': 'Single Shot',
+    '6': 'Line'
+}
+TRIGGER_INHIBITS = {
+    '0': 'Off',
+    '1': 'Triggers',
+    '2': 'AB',
+    '3': 'AB,CD',
+    '4': 'AB,CD,EF',
+    '5': 'AB,CD,EF,GH'
+}
+
+
+class DgChannel(BaseInterface, Device):
+    """
+    Delay generator single channel class
+
+    Parameters
+    ----------
+    prefix: str
+        Base PV for the delay generator channel
+
+    name: str
+        Alias of the channel
+    """
+    delay = Cpt(EpicsSignal, 'DelayAO', kind='hinted')
+    delay_str = Cpt(EpicsSignalRO, 'DelaySI', kind='normal')
+    reference = Cpt(EpicsSignal, 'ReferenceMO', kind='normal',
+                    doc='Reference channel from which the delay taken.')
+
+    tab_component_names = True
+    tab_whitelist = ['set', 'set_reference', 'get_str']
+
+    def get_str(self):
+        """ Returns the full <REF> + <DELAY> string. """
+        return self.delay_str.get()
+
+    def set(self, new_delay):
+        return set_and_wait(self.delay, new_delay)
```

Comment:
This will break `bluesky` scans of `DgChannel` devices, since `set` is no longer behaving correctly. `set` is supposed to return a status object immediately without waiting.
If you want `set` to pass through to the `delay` signal so `DgChannel` can be used as a settable/movable in a scan, something like this would work better:

```suggestion
    def set(self, new_delay, **kwargs):
        return self.delay.set(delay, **kwargs)
```
If you don't need to pass a full `DgChannel` into a `bluesky` scan you probably don't need to override the `set` method.
If you're intending to have a "wait for the value to change" interface it needs to be named something else than `set`.

Reply:
Thanks for your comment. I will go with what you suggest, as this is a variable we will want to scan. I've played a bit with the set, it is almost instantaneous, so the set_and_wait functionality does not really adds anything useful here.